src/lib/comms/sol-network-impl-linux.c

```diff
@@ -464,24 +464,24 @@ sol_network_subscribe_events(void (*cb)(void *data, const struct sol_network_lin
 
     SOL_NULL_CHECK(cb, false);
 
-    if (sol_network_start_netlink() < 0) return false;
+    if (sol_network_start_netlink() < 0) return -errno;
```

Comment:
Again, return the `errno` value is not correct in this case
I suggest that you do the following:
```C
int r;
r = sol_network_start_netlink();
SOL_INT_CHECK(r, < 0, r);
```